ui/pages/token-allowance/token-allowance.stories.js

```diff
@@ -0,0 +1,283 @@
+        {
+          op: 'add',
+          path: '/txParams/maxFeePerGas',
+          value: '0x4a817c800',
+        },
+        {
+          op: 'add',
+          path: '/txParams/maxPriorityFeePerGas',
+          value: '0x4a817c800',
+        },
+        {
+          op: 'replace',
+          path: '/loadingDefaults',
+          value: false,
+        },
+        {
+          op: 'add',
+          path: '/userFeeLevel',
+          value: 'custom',
+        },
+        {
+          op: 'add',
+          path: '/defaultGasEstimates',
+          value: {
+            estimateType: 'custom',
+            gas: '0xea60',
+            maxFeePerGas: '0x4a817c800',
+            maxPriorityFeePerGas: '0x4a817c800',
+          },
+        },
+      ],
+    ],
+    userFeeLevel: 'custom',
+    defaultGasEstimates: {
+      estimateType: 'custom',
+      gas: '0xea60',
+      maxFeePerGas: '0x4a817c800',
+      maxPriorityFeePerGas: '0x4a817c800',
+    },
+  },
+};
+
+// eslint-disable-next-line import/no-anonymous-default-export
+export default {
+  title: 'Pages/TokenAllowance',
+  argTypes: {
+    origin: {
+      control: 'text',
+    },
+    siteImage: {
+      control: 'text',
+    },
+    showCustomizeGasModal: {
+      action: 'showCustomizeGasModal',
+    },
+    useNonceField: {
+      control: 'boolean',
+    },
+    currentCurrency: {
+      control: 'text',
+    },
+    nativeCurrency: {
+      control: 'text',
+    },
+    ethTransactionTotal: {
+      control: 'text',
+    },
+    fiatTransactionTotal: {
+      control: 'text',
+    },
+    hexTransactionTotal: {
+      control: 'text',
+    },
+    isMultiLayerFeeNetwork: {
+      control: 'boolean',
+    },
+    supportsEIP1559: {
+      control: 'boolean',
+    },
+    userAddress: {
+      control: 'text',
+    },
+    tokenAddress: {
+      control: 'text',
+    },
+    data: {
+      control: 'text',
+    },
+    isSetApproveForAll: {
+      control: 'boolean',
+    },
+    setApproveForAllArg: {
+      control: 'boolean',
+    },
+    decimals: {
+      control: 'text',
+    },
+    dappProposedTokenAmount: {
+      control: 'text',
+    },
+    currentTokenBalance: {
+      control: 'text',
+    },
+    toAddress: {
+      control: 'text',
+    },
+    tokenSymbol: {
+      control: 'text',
+    },
+    txData: {
+      control: 'object',
+    },
+    warning: {
+      control: 'text',
+    },
+    fromAddressIsLedger: {
+      control: 'boolean',
+    },
+  },
+  args: {
+    ...defaultArgs,
+  },
+};
+
+const Template = (args) => <TokenAllowance {...args} />;
+
+export const DefaultStory = Template.bind({});
+
+DefaultStory.storyName = 'Default';
+
+const storeWithErrors = configureStore({
+  ...testData,
+  metamask: {
+    ...testData.metamask,
+    // TODO: Mock state correctly to show insufficient funds error
+    //
+    accounts: {
+      ...testData.metamask.accounts,
+      '0x9d0ba4ddac06032527b140912ec808ab9451b788': {
+        balance: '0x0',
+      },
+    },
+  },
```

Comment:
Hey @matthewwalsh0, do you have any ideas on how one could show the insufficient funds for gas error? I seem to be struggling with that

<img width="362" alt="Screenshot 2023-10-16 at 3 07 43 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/782210f1-1893-4a90-8ad8-0dab58dd805c">